src/app/CommandHandler.cpp

```diff
@@ -746,11 +766,31 @@ CHIP_ERROR CommandHandler::FinishStatus()
     return CHIP_NO_ERROR;
 }
 
+void CommandHandler::CreateBackupForResponseRollback()
+{
+    if (mState != State::NewResponseMessage && mState != State::AddedCommand)
+    {
+        return;
+    }
+    if (mInvokeResponseBuilder.GetInvokeResponses().GetError() != CHIP_NO_ERROR ||
+        mInvokeResponseBuilder.GetError() != CHIP_NO_ERROR)
+    {
+        return;
+    }
+    mInvokeResponseBuilder.Checkpoint(mBackupWriter);
+    mBackupState         = mState;
+    mRollbackBackupValid = true;
+}
+
 CHIP_ERROR CommandHandler::RollbackResponse()
 {
+    VerifyOrReturnError(mRollbackBackupValid, CHIP_ERROR_INCORRECT_STATE);
     VerifyOrReturnError(mState == State::Preparing || mState == State::AddingCommand, CHIP_ERROR_INCORRECT_STATE);
+    mInvokeResponseBuilder.GetInvokeResponses().ResetError();
+    mInvokeResponseBuilder.ResetError();
```

Comment:
Would a generate `ResetErrors()` method helps avoid these two?

Reply:
If it was done more than once I would consider it, but this is the only spot where we are calling `ResetError`, so it would take more code to create a generic `ResetErrors`